sample/src/main/java/no/finn/rxactivityresponse/sample/SimplePermissionButton.java

```diff
@@ -0,0 +1,55 @@
+package no.finn.rxactivityresponse.sample;
+
+import java.lang.ref.WeakReference;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import no.finn.android.rx.RxPermission;
+import no.finn.android.rx.RxPermissionRationale;
+import no.finn.android.rx.RxState;
+import no.finn.android.rx.RxStateRestart;
+
+import junit.framework.Assert;
+import rx.functions.Action1;
+
+public class SimplePermissionButton extends Button implements View.OnClickListener, RxStateRestart {
+    private final RxState rxState;
+
+    public SimplePermissionButton(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        // The State has to be created construction time (it's the only way we can get if we're serialized mid flow!)
+        rxState = RxState.get(context, ActivityResponses.GET_LOCATIONPERMISSION, new WeakReference<RxStateRestart>(this));
```

Comment:
Leaning towards Option 2 + Option 3. It's too easy for Option 1 to go wrong and leak a view/context without the developer having any warning whatsoever of what went wrong. Atleast with 2/3 you'll get an exception that makes sense (and one that should happen fairly consistently.. although depending on GC to clear the reference)